types/dom-chromium-ai/index.d.ts

```diff
@@ -0,0 +1,273 @@
+interface WindowOrWorkerGlobalScope {
+    readonly ai: AI;
+}
+
+interface AI {
+    readonly assistant: AIAssistantFactory;
+    readonly summarizer: AISummarizerFactory;
+    readonly writer: AIWriterFactory;
+    readonly rewriter: AIRewriterFactory;
+}
+
+interface AICreateMonitor {
+    ondownloadprogress: (event: DownloadProgressEvent) => void;
+
+    addEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: AddEventListenerOptions | boolean,
+    ): void;
+    removeEventListener(
+        type: "downloadprogress",
+        listener: (event: DownloadProgressEvent) => void,
+        options?: EventListenerOptions | boolean,
+    ): void;
+}
```

Comment:
I assume you did it this way instead of imeplementing/extending `EventTarget` because `EventTarget` isn't strictly typed?

Reply:
Oh, good catch! I missed extending from `EventTarget`. However, I still need to override the `addEventListener()`/`removeEventListener()` methods to provide more specific hints for the `downloadprogress`/`contextoverflow` events.